src/handwritten/_install_daqmx.py

```diff
@@ -0,0 +1,255 @@
+import contextlib
+import errno
+import importlib.resources as pkg_resources
+import json
+import logging
+import os
+import pathlib
+import subprocess
+import sys
+import tempfile
+import traceback
+import urllib.request
+from typing import Generator, List, Optional, Tuple
+import importlib.resources as pkg_resources
+
+import click
+
+if sys.platform.startswith("win"):
+    import winreg
+
+_logger = logging.getLogger(__name__)
+
+METADATA_FILE = "_installer_metadata.json"
+
+
+
+def _parse_version(version: str) -> Tuple[int, ...]:
+    """
+    Split the version string into a tuple of integers.
+
+    >>> _parse_version("23.8.0")
+    (23, 8, 0)
+    >>> _parse_version("24.0.0")
+    (24, 0, 0)
+    >>> _parse_version("invalid_version")
+    Traceback (most recent call last):
+    ...
+    click.exceptions.ClickException: Invalid version format found
+    """
+    try:
+        return tuple(int(part) for part in version.split("."))
+    except ValueError as e:
+        _logger.info("Failed to parse version.", exc_info=True)
+        raise click.ClickException(f"Invalid version format found:{str(e)}") from e
+
+
+def _get_daqmx_installed_version() -> Optional[str]:
+    """
+    Check for existing installation of NI-DAQmx.
+
+    """
+    try:
+        _logger.debug("Reading the registry entries to get installed DAQmx version")
+        with winreg.OpenKeyEx(
+            winreg.HKEY_LOCAL_MACHINE,
+            r"SOFTWARE\National Instruments\NI-DAQmx\CurrentVersion",
+            0,
+            winreg.KEY_READ | winreg.KEY_WOW64_32KEY,
+        ) as daqmx_reg_key:
+            product_name = winreg.QueryValueEx(daqmx_reg_key, "ProductName")[0]
+            product_version = winreg.QueryValueEx(daqmx_reg_key, "Version")[0]
+
+        if product_name == "NI-DAQmx":
+            _logger.info(
+                "Found registry entries for Product Name: %s and version %s",
+                product_name,
+                product_version,
+            )
+            return product_version
+    except FileNotFoundError:
+        _logger.info("No existing NI-DAQmx installation found.", exc_info=False)
+        return None
+    except PermissionError as e:
+        _logger.info("Failed to read the registry key.", exc_info=True)
+        raise click.ClickException(
+            f"Permission denied while trying to read the registry key"
+        ) from e
+    except OSError as e:
+        _logger.info("Failed to read the registry key.", exc_info=True)
+        raise click.ClickException(
+            f"An OS error occurred while trying to read the registry key"
+        ) from e
```

Comment:
User-facing errors should use user-facing terminology.

```suggestion
    except PermissionError as e:
        _logger.info("Failed to read the registry key.", exc_info=True)
        raise click.ClickException(
            f"Permission denied while getting the installed NI-DAQmx version."
        ) from e
    except OSError as e:
        _logger.info("Failed to read the registry key.", exc_info=True)
        raise click.ClickException(
            f"An OS error occurred while getting the installed NI-DAQmx version.\nDetails: {e}"
        ) from e
```